SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp

```diff
@@ -173,13 +173,21 @@ void USpatialConnectionManager::Connect(bool bInitAsClient, uint32 PlayInEditorI
 	bConnectAsClient = bInitAsClient;
 
 	const ISpatialGDKEditorModule* SpatialGDKEditorModule = FModuleManager::GetModulePtr<ISpatialGDKEditorModule>("SpatialGDKEditor");
-	if (SpatialGDKEditorModule != nullptr && SpatialGDKEditorModule->ShouldConnectToCloudDeployment() && bInitAsClient)
+	if (SpatialGDKEditorModule != nullptr && SpatialGDKEditorModule->ShouldConnectToCloudDeployment())
 	{
-		DevAuthConfig.Deployment = SpatialGDKEditorModule->GetSpatialOSCloudDeploymentName();
-		DevAuthConfig.WorkerType = SpatialConstants::DefaultClientWorkerType.ToString();
-		DevAuthConfig.UseExternalIp = true;
-		StartDevelopmentAuth(SpatialGDKEditorModule->GetDevAuthToken());
-		return;
+		if (bInitAsClient)
+		{
+			DevAuthConfig.Deployment = SpatialGDKEditorModule->GetSpatialOSCloudDeploymentName();
+			DevAuthConfig.WorkerType = SpatialConstants::DefaultClientWorkerType.ToString();
+			DevAuthConfig.UseExternalIp = true;
+			StartDevelopmentAuth(SpatialGDKEditorModule->GetDevAuthToken());
+			return;
+		}
+
+		if(SpatialGDKEditorModule->ShouldStartLocalServer())
```

Comment:
Perhaps we should also check that we are running on the server? I believe there is no need to set ReceptionistConfig.UseExternalIp to true on the client?

Reply:
If it is a client it will go in the previous if statement [if (bInitAsClient)] and the function will return, so only server should reach that if statement.

Reply:
Ahh I missed the return my mistake :p